File: mobile application development/front/react-native/src/api/userApi.ts
import { apiRequest } from './client';
import { API_PREFIX } from '../config/api';
import type { PortfolioResponse, UserMe } from './types';

export function getMeApi() {
  return apiRequest<UserMe>(`${API_PREFIX}/users/me`);
}

export function updateBalanceApi(balance: number) {
  return apiRequest<UserMe>(`${API_PREFIX}/users/me/balance`, {
    method: 'PUT',
    body: { balance },
  });
}

export function getPortfolioApi() {
  return apiRequest<PortfolioResponse>(`${API_PREFIX}/users/me/portfolio`);
}

File: mobile application development/front/kotlin/app/src/main/java/com/itmo/mybroker/api/UserApi.kt
package com.itmo.mybroker.api

object UserApi {
    private const val PREFIX = "/api/broker-app/v1"

    fun getMe(): UserMe = ApiClient.request("$PREFIX/users/me")

    fun updateBalance(balance: Double): UserMe =
        ApiClient.request("$PREFIX/users/me/balance", method = "PUT", body = mapOf("balance" to balance))

    fun getPortfolio(): PortfolioResponse = ApiClient.request("$PREFIX/users/me/portfolio")
}
